tools/shape_tool/ShapeWindow.cpp

```diff
@@ -0,0 +1,162 @@
+#include "../../../../lib/ui3/elements/TileShapeGraphic.cpp"
+
+const array<int> SHAPE_MAP = {
+	 8, 16,  0,  0,
+	 7, 15,  0,  0,
+	11,  3, 20, 17,
+	12,  4, 19, 18,
+	10,  9,  1,  2,
+	 6,  5, 13, 14
+};
+
+class ShapeWindow : Window
+{
+
+	private Container@ shape_container;
+	private Button@ custom_button;
+
+	private int _tile_shape = 0;
+	private int _selection_index = 2;
+
+	private bool _selecting_tiles = false;
+	private bool _using_custom_shape = false;
+
+	private TileShapes@ _selected_shape;
+
+	int tile_shape
+	{
+		get const { return _tile_shape; }
+		set
+		{
+			if(value < 0 or value > 20) return;
+			int index = SHAPE_MAP.find(value);
+			cast<Button>(shape_container.get_child(index)).selected = true;
+		}
+	}
+
+	bool selecting_tiles { get const { return _selecting_tiles; } }
+	bool using_custom_shape { get const { return _using_custom_shape; } }
+
+	ShapeWindow(UI@ ui)
+	{
+		super(ui, "Shape", false);
+
+		@layout = FlowLayout(ui, FlowDirection::Row);
+
+		@shape_container = Container(ui);
+
+		GridLayout@ grid = GridLayout(ui, 4);
+		grid.row_spacing = 0;
+		grid.column_spacing = 0;
```

Comment:
Was it this spacing that you wanted to be set to default?

Reply:
This seems much larger than standard:
![image](https://github.com/cmann1/adv-tools/assets/4889816/49d1b28c-f51a-40f3-8eb3-caf8045cc461)

Compared to the text tool e.g.
![image](https://github.com/cmann1/adv-tools/assets/4889816/07efd804-2806-4ffe-8dfb-e3d16adcf0bf)

Reply:
Ah ok looks like since some stuff needs a wrapper the default layout padding causes the space to double up.

Reply:
I've got the padding to *look* correct. Not sure if the way I did it is the best way. Had to fix a little bug in the ScrollView to get the Scrollbar to line up correctly (see the PR in DustScripts).